core/src/saros/git/GitManager.java

```diff
@@ -0,0 +1,108 @@
+package saros.git;
+
+import java.io.File;
+import java.io.IOException;
+import org.apache.log4j.Logger;
+import org.picocontainer.Startable;
+import saros.activities.GitCollectActivity;
+import saros.activities.GitRequestActivity;
+import saros.activities.GitSendBundleActivity;
+import saros.session.AbstractActivityConsumer;
+import saros.session.AbstractActivityProducer;
+import saros.session.IActivityConsumer;
+import saros.session.IActivityConsumer.Priority;
+import saros.session.internal.SarosSession;
+
+/**
+ * This Class is responsible for the negotiation between a user that want to send,receive and fetch
+ * commits.
+ *
+ * <p>In order to start the negotiation the user needs to call the method {@code
+ * sendCommitRequest()}.
+ *
+ * <p>Before starting negotiation the user need to set up his working directory tree by calling the
+ * method {@code changeWorkDirTree(File workDirTree)}. After calling it the first time the Manager
+ * hold a {@link JGitFacade} object.
+ */
+public class GitManager extends AbstractActivityProducer implements Startable {
+
+  private static final Logger log = Logger.getLogger(GitManager.class.getName());
+
+  private final SarosSession session;
```

Comment:
This is a leftover after I told him to not add the methods to the interface. I expected it would be clear that this also means that the methods will not appear in the SarosSession class ... but this is the current result.